src/shogun/features/DenseSubSamplesFeatures.h

```diff
@@ -174,6 +174,25 @@ template<class ST> class CDenseSubSamplesFeatures: public CDotFeatures
 	 * @param iterator as returned by get_feature_iterator
 	 */
 	virtual void free_feature_iterator(void* iterator);
+
+
+	/** does this class support compatible computation bewteen difference classes?
+	 * for example, this->dot(rhs_prt),
+	 * can rhs_prt be an instance of a difference class?
+	 *
+	 * @return whether this class supports compatible computation
+	 */
+	virtual bool support_compatible_class() const {return true;}
+
+	/** Given a class in right hand side, does this class support compatible computation? 
```

Comment:
The documentation should be a bit more clear here.

Returns whether this instance supports dot products /whatever with the feature class specified in the argument